src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -327,6 +332,30 @@ def _slice(
         return new_domain, slice_
 
 
+if dace:
+    # Extension of NdArrayField : Support for SDFGConvertible GT4Py Programs
+    @dataclasses.dataclass(frozen=True)
+    class NdArrayField(  # type: ignore # redefinition
```

Comment:
Just a comment: in this case it is not really needed to create a new subclass since it's not needed to add other parent classes, so we could just monkeypatch the original class with the two new extra methods...

Reply:
I would prefer to keep it with the subclass as I find it cleaner.
Although, if you think it is a better practice, I could change it.

Reply:
Eventually I followed your suggestion because the redefinition of the NdArrayField class was causing issues (some tests were failing).